src/Service/ProgrammesAggregationService.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Service;
+
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\ClipRepository;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\CoreEntityRepository;
+use BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain\CoreEntityMapper;
+use BBC\ProgrammesPagesService\Cache\CacheInterface;
+use InvalidArgumentException;
+
+class ProgrammesAggregationService extends AbstractService
+{
+    const AGGREGATION_VALID_TYPES = [
+        'Series',
+        'Episode',
+        'Clip',
+    ];
+
+    public function __construct(
+        CoreEntityRepository $repository,
+        CoreEntityMapper $mapper,
+        CacheInterface $cache
+    ) {
+        parent::__construct($repository, $mapper, $cache);
+    }
+
+    public function findChildrenByType(
+        array $findByAncestryIds,
```

Comment:
This should be the entity itself, and you should get the ancestry ids in this function